### ISSUE TITLE
Implementing Client-Side Translation Feature

### ISSUE THREAD

**Issue by CarlosUlisesOchoa** [svobik7/next-roots#88]
# Hello @svobik7,

I hope this message finds you well. I am currently using your "next-roots" project and I find it very helpful. I am however running into a challenge that I hope you can help with. I am trying to add a method to get the translations from a client-side component.

I need the ability to use a function like t('key.value') in the client-side components to manage translations. That would help me to manage translations better, for example I want to take advantage of the functionalities of: https://www.i18next.com/translation-function/formatting.

Thanks for your support.

**Comment by svobik7** [comment]
Hello @CarlosUlisesOchoa 

I am glad my lib helps you. Regarding `t()` I did not implement such a feature in my package because there were more mature solutions already out there. So I did not want to re-invent the wheel.

Anyway I would suggest to translate your app's texts in server side components and then pass the translations down to the client components by props. By doing that no extra JS is sent to client's browser.

Even I made custom `t()` function in example for `next-roots` (see https://github.com/svobik7/next-roots/blob/master/example/src/features/common/components/Page.tsx#L9 ) I recommend using more mature solution in real world projects. Libs like next-intl or next-i18next now supports app router (at least as beta) so I would use one of them.
